### PR TITLE
Editor: add a Guided Tour for editor basics

### PR DESCRIPTION
**Pull request by lsinger**
This PR adds a tour that gets triggered when a new user (i.e., account age is 7 days or less) enters the post editor (i.e., path starts with `/post/`). 

This is what it looks like:

editorbasicstour-v2

To test:
- Create a new user account, then head for the post editor a few times. Confirm that the tour gets triggered once and only once. 
- Go to `http://calypso.localhost:3000/post/SITE_URL?tour=editorBasicsTour` and confirm that the tour gets triggered. 
- Confirm that the tour's steps are worded appropriately. 

TODO:
- [x] add `translate` calls for the copy when the copy is final

This tour was made at the Cerberus meet-up. Props to @jerrysarcastic @dcoleonline @eurello @vparkhere @formosattic @mgalbritton. 🙌

### CODE REVIEW

**Comment by mtias** [comment]
This is cool. It'd be good to add one teaching how to collapse the sidebar for a content-focused view.

**Comment by lsinger** [comment]
> It'd be good to add one teaching how to collapse the sidebar

Thanks for you input @mtias! We've since changed the sidebar step to point to the cog. Do you think that plus a change in the copy would suffice? We want to keep the number of steps down as much as possible so as to decrease the probability of users dropping out over the course of the tour.

**Comment by mtias** [comment]
Definitely, with some copy that mentions you can collapse/expand on demand would be good.

**Comment by lsinger** [comment]
@mtias updated the tour — would love to have you take another look.

**Comment by kristastevens** [comment]
Hello and thanks for the ping! This is a great, succinct tour. Here are a few copy suggestions: 

> Welcome to the editor. Add a title here. 

Suggest: 

Welcome to the editor. Add a title to your post here. 

> Find additional settings in the sidebar, such as tags, categories, and the featured image. 

Suggest: 

Add tags, categories, and a featured image from the sidebar. Learn more about <a href="https://learn.wordpress.com/get-published/">how to publish a post</a>.

I had just updated this page at Learn on my last editor on call rotation five weeks ago. I'll update it again, as the settings have all moved to the right-hand side of the screen.

File: client/layout/guided-tours/style.scss [hunk 1]
```diff
@@ -1,6 +1,7 @@
 .guided-tours__step {
 	position: fixed;
 	max-width: 410px;
+	width: 410px;
```
Comment (mcsf) [inline client/layout/guided-tours/style.scss]: Could this come back to bite us later? Any other kind of step, or viewport, where we wouldn't want it? If we keep it, can we get rid of `max-width` right above?

File: client/layout/guided-tours/tours/editor-basics-tour.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+export const EditorBasicsTour = makeTour(
+	<Tour
+		name="editorBasicsTour"
+		version="20170321"
+		path="/post/"
+		when={ and(
+			isDesktop,
+			isNewUser,
+			) }
```
Comment (mcsf) [inline client/layout/guided-tours/tours/editor-basics-tour.js]: Minor: I'd keep the `when` in one line

File: client/layout/guided-tours/tours/editor-basics-tour.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+		>
```
Comment (mcsf) [inline client/layout/guided-tours/tours/editor-basics-tour.js]: Indentation: put `>` at the same level as `<Tour`. Same goes for the `Step` elements below.

File: client/layout/guided-tours/tours/editor-basics-tour.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+			target=".editor-title"
+			placement="below"
+			style={ { animationDelay: '5s', } }
+			>
+			<p>
+				Welcome to the editor! Add a title here.
+			</p>
+			<ButtonRow>
+				<Next step="write">Continue</Next>
+				<Quit>Quit</Quit>
```
Comment (mcsf) [inline client/layout/guided-tours/tours/editor-basics-tour.js]: Are the explicit labels for Continue and Quit needed?

File: client/layout/guided-tours/tours/editor-basics-tour.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+			name="publish"
+			arrow="right-top"
+			target=".editor-ground-control__publish-combo"
+			placement="beside"
+			style={ { marginTop: '-17px' } }
+			>
+			<p>
+				{
+					translate( 'Your changes are saved automatically. ' +
+									'Click {{strong}}Publish{{/strong}} to share your work with the world!',
```
Comment (mcsf) [inline client/layout/guided-tours/tours/editor-basics-tour.js]: This line should be indented two levels lower. :)

**Comment by eurello** [comment]
Maybe for: 

> Add tags, categories, and a featured image from the sidebar. Learn more about how to publish a post.

Change to: 

>  Add tags, categories, a featured image, and more from the sidebar. Learn more about how to publish a post.

(Just because there are other options there.)

File: client/layout/guided-tours/style.scss (hunk 1 above)
Comment (lsinger) [inline client/layout/guided-tours/style.scss]: Thanks for the pointer. I looked at this a bit more and have now moved the width setting from the `guided-tours__first-step` class up here. This solves the problem we had here (very narrow steps when little text present) and at the same time reuses what we'd already been using anyway.

**Comment by lsinger** [comment]
Thanks @mcsf for your comments! I've addressed them in c36bfa2. Could you take another look to see whether this is good to go, code-wise?

**Comment by lsinger** [comment]
Thanks for your suggestions, @kristastevens! 

> Welcome to the editor. Add a title to your post here.

This can run in both a post or page editor, so I'll keep this as-is. 

> Add tags, categories, and a featured image from the sidebar. Learn more about how to publish a post.

I've split this suggestion up and used it in the last two steps. First comes the "tags, categories, and ..." sentence, and then in the final step we had the "what to learn more?" link. Here's a GIF of the updated flow:

editorbasicstour-v2

**Comment by kristastevens** [comment]
This looks great, @lsinger!

**Comment by folletto** [comment]
Tested, looks REALLY GOOD. I think it's a great use for the Guided Tour. :)

---

I'd just mention, if you think it's possible and doesn't disrupt this, to highlight that the `+` button does more than just images, as we'd like to build more on that UI interaction in the future.

screen shot 2017-05-02 at 17 00 55

No new steps, something short that keeps emphasis still on images:

> Click the `+` to add images, `\/` to add even more things.

**Comment by marekhrabe** [comment]
The tour looks great - really nice introduction for new users. There is one thing that we might want to fix before merging:

When I strictly follow steps, there is one that tells me to click the + to add images. When I do just that, it opens the modal but it shows under the tour bubble for me. I was able to replicate this every time I tried. The arrow keeps pointing to the + button that is under the modal.

screen shot 2017-05-03 at 14 04 55

I can also see one tiny problem with the popover on the very same button. I understand this one might not be worth fixing due to complexity.

popover

**Comment by marekhrabe** [comment]
Just a tiny bit of fiddling with the devtools let me a solution to the first problem: setting an explicit `z-index` on that particular step to something low can make it stay under the modal overlay without breaking anything else for us.

**Comment by lsinger** [comment]
> I'd just mention, if you think it's possible and doesn't disrupt this, to highlight that the + button does more than just images, as we'd like to build more on that UI interaction in the future.

Thanks for that reminder, @folletto! I'm definitely up for adjusting this once we have more behind that dropdown. As it is I think we should hold off for now, as the features aren't there yet. 

> When I strictly follow steps, there is one that tells me to click the + to add images. When I do just that, it opens the modal but it shows under the tour bubble for me. 

Thanks @marekhrabe, great catch. Fixed in 24ca66e.

**Comment by lsinger** [comment]
Well, **actually**, this tour only gets triggered when editing a post, never when editing a page. Sorry @kristastevens for getting that wrong.

**Comment by kristastevens** [comment]
No worries, @lsinger -- thanks so much for clarifying!

File: client/layout/guided-tours/tours/editor-basics-tour.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+			arrow="top-left"
+			target=".mce-toolbar-grp.mce-container"
+			placement="below"
+			style={ { marginTop: '40px' } }
+		>
+			<p>
+				{ translate( 'Write your post in the content area.' ) }
+			</p>
+			<img
+				src="https://en-support.files.wordpress.com/2017/03/editor-content-area_360.gif"
```
Comment (mcsf) [inline client/layout/guided-tours/tours/editor-basics-tour.js]: I recommend going to the relevant Media Library to find these resources and cross-reference either this PR or this file (`client/layout/guided-tours/tours/editor-basics-tour.js`), to minimize risks of having these resources deleted by someone unaware of their use in Calypso.

**Comment by gemmagarner** [comment]
Nice work!

'Add images' is cut off at the top of the browser for me:

guided-tour-cutoff

Based on the video above, 'Write your posts' also sits a little higher than it should:

guided-tour-cutoff-2

Using Chrome v58.0.3029.96 (Mac).

File: client/layout/guided-tours/tours/editor-basics-tour.js (hunk 6 above)
Comment (lsinger) [inline client/layout/guided-tours/tours/editor-basics-tour.js]: @jerrysarcastic I think I don't have access to that blog / media library? Would you be up to add a note there?

Comment (lsinger) [inline client/layout/guided-tours/tours/editor-basics-tour.js]: @mcsf gave me access. I added a note.

**Pull request merged**